javascript/src/components/streamrules/StreamRuleForm.jsx

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+var React = require('react/addons');
+var BootstrapModal = require('../bootstrap/BootstrapModal');
+var Input = require('react-bootstrap').Input;
+var Bubble = require('../support/Bubble');
+var HumanReadableStreamRule = require('./HumanReadableStreamRule');
+
+var StreamRuleForm = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    getInitialState() {
+        return this.props.streamRule;
+    },
+    getDefaultProps() {
+        return {
+            streamRule: {field: "", type: 1, value: "", inverted: false}
+        };
+    },
+    _resetValues() {
+        this.setState(this.props.streamRule);
+    },
+    _onSubmit(evt) {
+        if (this.state.type === 5) {
+            this.state.value = "";
+        }
+        this.props.onSubmit(this.props.streamRule.id, this.state);
+        this.refs.modal.close();
+    },
+    _formatStreamRuleType(streamRuleType) {
+        return <option key={'streamRuleType'+streamRuleType.id} value={streamRuleType.id}>{streamRuleType.short_desc}</option>;
+    },
+    open() {
+        this._resetValues();
+        this.refs.modal.open();
+    },
+    close() {
+        this.refs.modal.close();
+    },
+    render() {
+        var streamRuleTypes = this.props.streamRuleTypes.map(this._formatStreamRuleType);
+        var valueBox = (this.state.type !== 5 ? <Input type='text' required={true} label='Value' placeholder='19983' valueLink={this.linkState('value')}/> : "");
+        return (
+            <BootstrapModal ref='modal' onCancel={this.close} onConfirm={this._onSubmit} cancel="Cancel" confirm="Save">
+                <div>
+                    <h2>{this.props.title}</h2>
+                </div>
+                <div>
+                    <div className='col-md-8'>
+                        <Input type='text' required={true} label='Field' placeholder='user_id' valueLink={this.linkState('field')}/>
```

Comment:
Using `user_id` as placeholder feels really weird, as it looks like something already wrote in the input. Can we use something else or even leave the placeholder out? Same goes for the value input field.